### ISSUE TITLE
crash with -p <url>

### ISSUE THREAD

**Issue by matthiaskrgr** [servo/servo#9927]
If the interval to the -p argument is forgotten, servo crashes.
````
./mach run --dev  -p  "http://www.example.com"
thread '<main>' panicked at 'Error parsing option: -p: ParseFloatError { kind: Invalid }', ../src/libcore/result.rs:746
stack backtrace:
   1:     0x556832951780 - sys::backtrace::tracing::imp::write::h578d6e47592a4041cHv
   2:     0x556832955b2b - panicking::default_handler::_$u7b$$u7b$closure$u7d$$u7d$::closure.44995
   3:     0x556832955798 - panicking::default_handler::hffa00c845e94c6581nA
   4:     0x55683293f24c - sys_common::unwind::begin_unwind_inner::hd824946792143cc8vvu
   5:     0x55683293f6d8 - sys_common::unwind::begin_unwind_fmt::h3777a045191296c5Buu
   6:     0x556832950bf1 - rust_begin_unwind
   7:     0x55683298cbcf - panicking::panic_fmt::h548fe28297c9d83b7mM
   8:     0x556831d64a71 - result::unwrap_failed::h2098280681605981451
                        at ../src/libcore/macros.rs:29
   9:     0x556831d6959f - result::Result<T, E>::expect::h17045132547580255910
                        at ../src/libcore/result.rs:708
  10:     0x556831d6951a - opts::from_cmdline_args::_$u7b$$u7b$closure$u7d$$u7d$::closure.22759
                        at /home/matthias/vcs/github/servo_dev/components/util/opts.rs:610
  11:     0x556831d693a2 - option::Option<T>::map::h16986006466550967684
                        at ../src/libcore/option.rs:390
  12:     0x556831d565c9 - opts::from_cmdline_args::h01255006a4429c20Uad
                        at /home/matthias/vcs/github/servo_dev/components/util/opts.rs:609
  13:     0x55682f8923b3 - main::hb2ad826330afabd3Naa
                        at /home/matthias/vcs/github/servo_dev/components/servo/main.rs:53
  14:     0x5568329553f4 - sys_common::unwind::try::try_fn::h17074903832088339578
  15:     0x556832950b7b - __rust_try
  16:     0x556832954e8b - rt::lang_start::hc48b7676a10bcd987fA
  17:     0x55682f997c89 - main
  18:     0x7f823134957f - __libc_start_main
  19:     0x55682f859b98 - _start
  20:                0x0 - <unknown>
Servo exited with return value 101
````
However
````
./mach run --dev   "http://www.example.com" -p 
````
works just fine.

Perhaps an error message could be printed instead of crashing?

@ ebe7d9ebfb28d484ff0cce4e70cb8809f5867af5

**Comment by jdm** [comment]
To be fair, it did print an error: "Error parsing option: -p: ParseFloatError { kind: Invalid }" However, we could conceivably do better than just calling `expect` on such things: https://github.com/servo/servo/blob/ebe7d9ebfb28d484ff0cce4e70cb8809f5867af5/components/util/opts.rs#L608

**Comment by KiChjang** [comment]
Is this fixed by #10221?
